File: apps/desktop/src/lib/desktop-api/journal.ts
// Journal desktop API.

import { invoke } from "@tauri-apps/api/core";

/**
 * List journal entries for a workspace.
 * Note: Backend currently returns an empty array (placeholder).
 * @param workspaceId - The workspace ID to list entries for
 */
export async function listJournalEntries(workspaceId: string): Promise<string[]> {
  return invoke("list_journal_entries", { workspaceId });
}

File: apps/desktop/src-tauri/src/commands/journal.rs
// Journal Tauri commands — Phase 1 placeholder.
// Note: Thesis commands have been moved to commands::thesis module.

use crate::error::AppError;

#[tauri::command]
pub async fn list_journal_entries(workspace_id: String) -> Result<Vec<String>, AppError> {
    // TODO: Implement journal entries retrieval from database
    // Currently returns empty array as placeholder
    let _ = workspace_id; // Acknowledge parameter until implementation
    Ok(Vec::new())
}
